packages/analytics-js-integrations/src/integrations/CommandBar/browser.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable class-methods-use-this */
+import {
+  NAME,
+  DISPLAY_NAME,
+} from '@rudderstack/analytics-js-common/constants/integrations/CommandBar/constants';
+import Logger from '../../utils/logger';
+import { loadNativeSdk } from './nativeSdkLoader';
+
+const logger = new Logger(DISPLAY_NAME);
+
+class CommandBar {
+  constructor(config, analytics, destinationInfo) {
+    if (analytics.logLevel) {
+      logger.setLogLevel(analytics.logLevel);
+    }
+    this.analytics = analytics;
+    this.orgId = config.orgId;
+    this.name = NAME;
+    ({
+      shouldApplyDeviceModeTransformation: this.shouldApplyDeviceModeTransformation,
+      propagateEventsUntransformedOnError: this.propagateEventsUntransformedOnError,
+      destinationId: this.destinationId,
+    } = destinationInfo ?? {});
+  }
+
+  init() {
+    loadNativeSdk(this.orgId);
+  }
+
+  isLoaded() {
+    return !!(window.CommandBar && typeof window.CommandBar === 'object');
+  }
+
+  isReady() {
+    return this.isLoaded();
+  }
+
+  /**
+   * Idenitfies a user with userId and traits
+   * Commandbar doc: https://www.commandbar.com/docs/sdk/lifecycle/boot/
+   * @param {*} rudderElement
+   * @returns
+   */
+  identify(rudderElement) {
+    const { message } = rudderElement;
+    const { traits, userId, context } = message;
+    const userTraits = { ...context.traits, ...traits };
+    const { hmacId } = userTraits;
+    if (!userId) {
+      logger.error('userId from identify call is missing');
+      return;
+    }
+
+    const instanceAttributes = {}; // stores commandBar options associated with the current session.
+
+    if (hmacId) {
+      instanceAttributes.hmac = hmacId;
+      delete userTraits.hmacId;
+    }
+    window.CommandBar.boot(userId, userTraits, instanceAttributes);
```

Comment:
Should we call `boot` twice if a user is identified twice but the second time with only additional traits?

I believe booting should happen only once. Correct me if I'm wrong.

Reply:
Yes booting should happen only once.
But can't that be handled at implementation level.
Also how will we knoe if a particular user is being identified or not. Shall we use cookie for this ?

Reply:
Users can make mistakes. However, let's check with the CB team what happens when we boot with the same user ID but a different (extra) set of traits.

If this is a problem, then we can store if a user was identified and handle it accordingly.

Reply:
Starting a thread with CB team

Reply:
Confirmed with CB team it is update or create function hence we can call multiple times